### ISSUE TITLE
keras model with multiple losses and outputs +morph_net

### ISSUE THREAD

**Issue by itabhiyanta** [google-research/morph-net#47]
Hi

Thanks a lot for open sourcing the morph-net code. It is something i really wish to use since i spent a lot of time trying to get my network to train last month for the RL problem i am working on. I have a model in Keras with multiple losses passed in a dict. Now if i wish to add e.g. a flops-regularizer to my model I wrote something like the following:

        state_input = Input(shape=(self.board_size, self.board_size+self.grid_size, 
        self.stack_frame_size,))
        actual_value = Input(shape=(1,))
        predicted_value = Input(shape=(1,))
        #adv = Input(shape=(1,))
        old_prediction_a = Input(shape=(self.output_size,))
        old_prediction_b = Input(shape=(self.output_size,))
        old_prediction_c = Input(shape=(self.output_size,))
        old_prediction_d = Input(shape=(self.output_size,))

        c1 = Conv2D(32, 10, strides=9, activation='relu', name='Conv1')(state_input)#6
        flat = Flatten()(c1)
        d1 = Dense(2048, activation='relu', name='Dense1')(flat)
        logits_a = Dense(self.output_size, activation = 'softmax',name='logits_a')(d1)
        logits_b = Dense(self.output_size, activation = 'softmax', name='logits_b')(d1)
        logits_c = Dense(self.output_size, activation = 'softmax', name='logits_c')(d1)
        logits_d = Dense(self.output_size, activation = 'softmax', name='logits_d')(d1)

        model = Model(inputs=[state_input, actual_value, predicted_value,
                              old_prediction_a, old_prediction_b,
                              old_prediction_d,old_prediction_c],
                              outputs=[logits_a, logits_b, logits_c, logits_d])
        model.load_weights(weights_path)

        
        network_regularizer = flop_regularizer.GammaFlopsRegularizer(
            [logits_a.op, logits_b.op, logits_c.op, logits_d.op], gamma_threshold=1e-3)
        self.actor_exporter = structure_exporter.StructureExporter(network_regularizer.op_regularizer_manager)
        regularization_strength = 1e-10
        regularizer_loss = (network_regularizer.get_regularization_term() * regularization_strength)
        losses = {
            'logits_a': ppoloss(actual_value=actual_value, predicted_value=predicted_value,
                                                           old_prediction=old_prediction_a),
            'logits_b': ppoloss(actual_value=actual_value, predicted_value=predicted_value,
                                                           old_prediction=old_prediction_b),
            'logits_c': ppoloss(actual_value=actual_value, predicted_value=predicted_value,
                                                           old_prediction=old_prediction_c),
            'logits_d': ppoloss(actual_value=actual_value, predicted_value=predicted_value,
                                                           old_prediction=old_prediction_d),
        }
        model.compile(optimizer=Adam(lr=1e-4),
                      loss=[losses])
`
If i add regularizer loss to my dict i get an error that since my outputs number 4 I must have only 4 losses. I cannot add it to the losses since then i have a problem that losses is a dict and regularization_loss is a float.
How may I add the loss in my particular problem

**Comment by itabhiyanta** [comment]
i went back to tensorflow and actually the example code compiles and works although the regularization loss and cost are flat.

**Comment by rodrigo2019** [comment]
Could you share your code?
